### PR TITLE
Display matrix TaskRuns on the PipelineRun details page

### PR DESCRIPTION
**Pull request by AlanGreene**
# Changes

Related to https://github.com/tektoncd/dashboard/issues/2410

Add support for displaying all TaskRuns produced by a matrix task on the PipelineRun details page.

Future updates will improve the display of the matrix information making it easier to differentiate between them on the left task/step nav, and making it clearer which params are provided from the matrix.

/kind feature

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (new features, significant UI changes, API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
Add support for displaying TaskRuns for a matrix task on the PipelineRun details page.
This allows users to access logs, status, and other details of the matrix-generated TaskRuns that were not previously available on the PipelineRun details page. Future releases will improve the display of this information.
```

### CODE REVIEW

**Comment by AlanGreene** [comment]
Hold for a bit so we can publish a new version of the packages with https://github.com/tektoncd/dashboard/pull/3082. I'll unhold once that's done in the next hour or so.
/hold

**Comment by AlanGreene** [comment]
Ready for review
/hold cancel

File: packages/components/src/components/TaskTree/TaskTree.jsx [hunk 1]
```diff
@@ -58,10 +66,17 @@ const TaskTree = ({
         const { reason, status } = getStatus(taskRunToUse);
         const { steps } = taskRunToUse.status || {};
         const expanded =
-          (!selectedTaskId && erroredTask?.metadata.uid === uid) ||
-          selectedTaskId === pipelineTaskName ||
-          (!erroredTask && !selectedTaskId && index === 0);
-        const selectDefaultStep = !selectedTaskId;
+          !hasExpandedTask &&
```
Comment (briangleeson) [inline packages/components/src/components/TaskTree/TaskTree.jsx:69]: Found this value tough to deconstruct. Perhaps worth adding a comment or two, or splitting up into more variables with explanatory names?

**Review by briangleeson — COMMENTED** [review]
Looks good overall, just one suggestion

File: packages/components/src/components/TaskTree/TaskTree.jsx (hunk 1 above)
Comment (AlanGreene) [reply packages/components/src/components/TaskTree/TaskTree.jsx:71]: Yeah there's a lot going on there, added comments for each branch. Let me know if that helps

Comment (briangleeson) [reply packages/components/src/components/TaskTree/TaskTree.jsx:71]: Looks good, thanks

**Review by briangleeson — APPROVED** [review]
/lgtm

**Pull request merged**